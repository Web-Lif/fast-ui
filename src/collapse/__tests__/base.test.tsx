import React from 'react'
import { render } from '@testing-library/react'
import Collapse from '../index'

const TestCollapse = () => {
    const text = `
        A dog is a type of domesticated animal.
        Known for its loyalty and faithfulness,
        it can be found as a welcome guest in many households across the world.
    `
    return (
        <div
            style={{
                height: 600,
            }}
        >
            <Collapse defaultActiveKey={['1']}>
                <Collapse.Panel header="This is panel header 1" key="1">
                    <p>{text}</p>
                </Collapse.Panel>
                <Collapse.Panel header="This is panel header 2" key="2">
                    <p>{text}</p>
                </Collapse.Panel>
                <Collapse.Panel header="This is panel header 3" key="3">
                    <p>{text}</p>
                </Collapse.Panel>
            </Collapse>
        </div>
    )
}

test('snapshot', async () => {
    const { container } = render(<TestCollapse />)
    expect(container).toMatchSnapshot()
})
