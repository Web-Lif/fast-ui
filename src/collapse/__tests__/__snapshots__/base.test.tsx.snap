// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    style="height: 600px;"
  >
    <div
      class="ant-collapse ant-collapse-icon-position-start css-4ruiw4 css-dev-only-do-not-override-1e67qka"
    >
      <div
        class="ant-collapse-item ant-collapse-item-active"
      >
        <div
          aria-disabled="false"
          aria-expanded="true"
          class="ant-collapse-header"
          role="button"
          tabindex="0"
        >
          <div
            class="ant-collapse-expand-icon"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                style="transform: rotate(90deg);"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </div>
          <span
            class="ant-collapse-header-text"
          >
            This is panel header 1
          </span>
        </div>
        <div
          class="ant-collapse-content ant-collapse-content-active"
        >
          <div
            class="ant-collapse-content-box"
          >
            <p>
              
        A dog is a type of domesticated animal.
        Known for its loyalty and faithfulness,
        it can be found as a welcome guest in many households across the world.
    
            </p>
          </div>
        </div>
      </div>
      <div
        class="ant-collapse-item"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="ant-collapse-header"
          role="button"
          tabindex="0"
        >
          <div
            class="ant-collapse-expand-icon"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </div>
          <span
            class="ant-collapse-header-text"
          >
            This is panel header 2
          </span>
        </div>
      </div>
      <div
        class="ant-collapse-item"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="ant-collapse-header"
          role="button"
          tabindex="0"
        >
          <div
            class="ant-collapse-expand-icon"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </div>
          <span
            class="ant-collapse-header-text"
          >
            This is panel header 3
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
